pkg/controller/controller_flow_instance_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/kubernetes-sigs/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pmorie/go-open-service-broker-client/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestProvisionInstanceWithRetries tests creating a ServiceInstance
+// with retry after temporary error without orphan mitigation.
+func TestProvisionInstanceWithRetries(t *testing.T) {
```

Comment:
how you ensure here that there was no orphan mitigation?
Shouldn't you check calls made to Service Broker? Without that u cannot be sure if the orphan wasn't triggered

Reply:
I've enhanced the test to cover with and without orphan mitigation, an assert for deprovisioning calls added to be sure if the orphan mitigation was processed or not